Grey and purple slime crystal fixes

Ravellon opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Right now, instead of using add_nutrition() function the grey crystal directly adjusts slime's nutrition. This prevents slimes from having their life effects triggered on nutrition change, such as splitting or raising their charge, while making them unwilling to feed once nutrition rises above 1000. That makes them unable to split once they cross that threshold.

With this change, grey crystal will feed slimes to 996/1200 nutrition and 9/10 growth instead of overfeeding them.
(if the conditional checks are removed the slimes will be fed infinitely and will split at getting to 10/10 growth, which will result in infinite slime reproduction. Not a good idea, I don't think)

Also, purple crystals now will heal toxin damage for slime races instead of dealing it.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

This way, leaving slimes in range of crystal for some time will put them into perfect range for either breeding or crossbreeding while not risking infinite slime spawning.
And it's just silly that xenobio healing crystals slowly kill xenobio sourced races.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
fix: Grey slime crystals will now slowly feed the slimes in range to 996/1200 nutrition and 9/10 growth range, instead of overfeeding them and making them unable to split.
fix: Purple slime crystals will now heal toxin damage for slime races instead of dealing it.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Ravellon committed:
Grey slime crystal fix

Grey crystal will feed slimes to 996/1200 nutrition and 9/10 growth instead of overfeeding them and making them unable to split on their own.

utilbird commented on code/modules/research/xenobiology/crossbreeding/_structures.dm:
```diff
@@ -119,7 +119,8 @@ GLOBAL_LIST_EMPTY(bluespace_slime_crystals)
 		return
 	var/mob/living/simple_animal/slime/slime_mob = affected_mob
-	slime_mob.nutrition += 2
+	if((slime_mob.nutrition < 996)||(slime_mob.amount_grown < (SLIME_EVOLUTION_THRESHOLD - 1)))
```
Brackets aren't needed
```suggestion
	if(slime_mob.nutrition < 996) || slime_mob.amount_grown < SLIME_EVOLUTION_THRESHOLD - 1)
```

Ravellon committed:
Update code/modules/research/xenobiology/crossbreeding/_structures.dm

Co-authored-by: TheFakeElon <59686430+TheFakeElon@users.noreply.github.com>

utilbird commented on code/modules/research/xenobiology/crossbreeding/_structures.dm:
```diff
@@ -119,7 +119,8 @@ GLOBAL_LIST_EMPTY(bluespace_slime_crystals)
 		return
 	var/mob/living/simple_animal/slime/slime_mob = affected_mob
-	slime_mob.nutrition += 2
+	if(slime_mob.nutrition < 996) || slime_mob.amount_grown < SLIME_EVOLUTION_THRESHOLD - 1)
```
Ah crap, messed up the initial code suggestion lmao, my bad

```suggestion
	if(slime_mob.nutrition < 996 || slime_mob.amount_grown < SLIME_EVOLUTION_THRESHOLD - 1)
```

Ravellon committed:
Update code/modules/research/xenobiology/crossbreeding/_structures.dm

Co-authored-by: TheFakeElon <59686430+TheFakeElon@users.noreply.github.com>

Ravellon commented:
I missed it too, so it's my fault for not being attentive enough.

Having no experience with BYOND I just do what has to work for sure no matter the language.

Ravellon committed:
Update _structures.dm

Also, purple crystals now heal toxin damage for slime people instead of dealing it.

Ravellon commented:
On a related note. The purple crystal is very very weak. It has 1/10 chance to heal 2 damage of any basic type or 1/10 chance to heal 2 damage to heart, lungs, liver or brain.
And it does not matter if there is no damage of the chosen type, it tries to heal it anyway and wastes the healing tick.

This is just garbage, tbh.

Rukofamicom commented:
> This prevents slimes from having their life effects triggered on nutrition change, such as splitting

This is not true at all. Slimes fed by the crystal grow and reproduce just the same as normal. I tested this crystal by putting it in an enclosure with one slime while doing all the rest of the bugtesting. By the time I was done with the rest of the bugtesting, the pen had 8-12 slimes per tile.

No electric charge is probably correct though. Not being able to feed is negligible since they divide anyway. Are you certain you hadn't reached the entity limit when you had your experience? This will prevent slimes from feeding and reproducing

Ravellon commented:
This is very weird. I managed to reproduce my issue once. The other three times it was as you described.
As in, one time they stuck at under 10 growth and got overfed, making them unable to split, even after I culled their numbers.
The other three times they continued to split after I culled the numbers.

Closing for now.
